include/ck/utility/inner_product.hpp

```diff
@@ -112,6 +124,19 @@ __device__ void inner_product<half2_t, half2_t, float>(const half2_t& a, const h
 #endif
 }
 
+template <>
+__device__ void inner_product<bhalf4_t, bhalf4_t, float>(const bhalf4_t& a, const bhalf4_t& b, float& c)
+{
+    const vector_type<bhalf_t, 4> a_vector{a};
+    const vector_type<bhalf_t, 4> b_vector{b};
+
+    static_for<0, 2, 1>{}([&](auto i) {
+        inner_product(a_vector.AsType<half2_t>()[i],
```

Comment:
Should be faster but probably with less precision than AsType <float2>

Reply:
Yeah I think I have a typo here. Thanks!